common/script/script-api/src/main/java/org/thingsboard/script/api/js/AbstractJsInvokeService.java

```diff
@@ -93,6 +95,21 @@ protected void doRelease(UUID scriptId) throws Exception {
         doRelease(scriptId, scriptInfoMap.remove(scriptId));
     }
 
+    @Override
+    public ListenableFuture<UUID> eval(TenantId tenantId, ScriptType scriptType, String scriptBody, String... argNames) {
+        if (!isExecEnabled(tenantId)) {
+            return error("Script Execution is disabled due to API limits!");
+        }
+        if (scriptBodySizeExceeded(scriptBody)) {
+            return error(format("Script body exceeds maximum allowed size of %s symbols", getMaxScriptBodySize()));
+        }
```

Comment:
I don't like the copy-paste. Introduce method `validate(String scriptBody)` in the `AbstractScriptInvokeService` with no implementation by default and extend it in the JS service. Then call in the `AbstractScriptInvokeService#eval`